sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -40,20 +60,7 @@ def __init__(self, credential, *scopes, **kwargs):  # pylint:disable=unused-argu
     @staticmethod
     def _enforce_https(request):
         # type: (PipelineRequest) -> None
-
-        # move 'enforce_https' from options to context so it persists
-        # across retries but isn't passed to a transport implementation
-        option = request.context.options.pop("enforce_https", None)
-
-        # True is the default setting; we needn't preserve an explicit opt in to the default behavior
-        if option is False:
-            request.context["enforce_https"] = option
-
-        enforce_https = request.context.get("enforce_https", True)
-        if enforce_https and not request.http_request.url.lower().startswith("https"):
-            raise ServiceRequestError(
-                "Bearer token authentication is not permitted for non-TLS protected (non-https) URLs."
-            )
+        return _enforce_https(request)
```

Comment:
Are you sure you want to use the exactly same method name? Looks confusing. :)

Reply:
Hmmm I hear you but the good name is taken 🤣 

🤷 `_enforce_https_impl`?